packages/server/src/routes/tenants.js

```diff
@@ -17,6 +17,28 @@ router.get('/', requireUser, async (req, res) => {
     res.json(response);
 });
 
+router.post('/', requireAdminUser, async (req, res) => {
+    if (!req.body.name || !req.body.agency_id) {
+        res.status(400).send('Tenant name and agency ID is required');
+        return;
+    }
+
+    try {
+        const tenant = {
+            display_name: req.body.name,
+            main_agency_id: req.body.agency_id,
```

Comment:
i might be getting this confused, since this tenant entity is new to me. A tenant is the main entity and agencies are then created under that tenant, correct? So I think we will not have agency_id.

1. Create Nevada tenant
2. Create Grant Office tenant
3. Update Nevada tenant as the main agency

Im not entirely sure what is the usage of main_agency_id

Reply:
the agencies are in a tree hierarchy under the tenant. `main_agency_id` is the root of the tree - the top-level agency that belongs to the tenant. This agency is required when the tenant is first created.

Reply:
@mhuang-usdr I'm a little confused - if the agency belongs to the tenant, couldn't it be possible for a tenant to be created before its agency is created?

Reply:
@slants3 I apologize, I realize forgot to explain this earlier!

The requirements for the app to function as expected are:
- Every tenant must have at least one agency - the main agency 
- Every agency must have at least one user, and they must be an admin.

Thus the requirements for spinning up a new tenant are:
- A new tenant in the tenant database
- A new agency assigned to the new tenant ("Grants Office" by default)
- A new admin user ("grants.dev@usdigitalresponse.org" by default) assigned to that tenant

I'm not sure what the best order or best practices to create those entries are, that's up to the eng team!


(Also, a possible snag - iirc users _might_ still be keyed by their email. In which case having the same user (grants.dev@usdigitalresponse.org) across multiple tenants will cause issues. The desired experience is that we _are_ able to have the same user in multiple tenants, but in the interim you can use "grants.dev+nv@usdigitalresponse.org" for the Nevada tenant.

Reply:
Makes total sense, thanks for explaining that!

Reply:
@slants3 we should create a new agency when we create a new tenant. Going forward, we should never need to reuse an agency for a new tenant. 

That is only happening right now due to a database cleanup (there are a lot of things in there that got really messed up)

Reply:
I see - what is the workflow for the agency + tenant creation? I'm trying to understand if they should be happening synchronously or if the new agency can be created first and then just passed into the tenant creation as an `main_agency_id`.

Reply:
@slants3 no current existing workflow. The agency should be created as the tenant is created.  Here's an example user journey.

Mindy, the PM, wants to onboard a new tenant. She runs the script (or, when we have a UI down the road, clicks the button), and a new tenant will be automatically set up with the main agency and default admin user (grants.dev@usdigitalresponse.org)

Reply:
> Just ran across #193 -- is this intended to supersede that, or will we have _both_ UI for this and a script?

Referencing this - I think in the immediate term we're going to use a script, but in the long term we will eventually want this to be an UI element